app/src/main/java/com/arctouch/codechallenge/ui/home/HomeActivity.kt

```diff
@@ -0,0 +1,50 @@
+package com.arctouch.codechallenge.ui.home
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProviders
+import android.databinding.DataBindingUtil
+import android.os.Bundle
+import android.support.annotation.StringRes
+import android.support.design.widget.Snackbar
+import android.support.v7.app.AppCompatActivity
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.databinding.HomeActivityBinding
+import com.arctouch.codechallenge.util.recycler_view.EndlessRecyclerOnScrollListener
+
+class HomeActivity : AppCompatActivity() {
+    private lateinit var binding: HomeActivityBinding
+    private lateinit var viewModel: HomeViewModel
+
+    private var pagesLoaded: Int = 1
```

Comment:
Given this property is in the `Activity`, how would you (unit) test the business rule that, when the user scrolls to the bottom of the list, a new page must be loaded?

Reply:
By comparing a mock pages loaded with the "totalPages" property that comes in the movie list in the "upcomingMovies" request and seeing if the pagesLoaded have reached the totalPages.